pkg/operator/ceph/cluster/controller.go

```diff
@@ -981,3 +1011,77 @@ func validateExternalClusterSpec(cluster *cluster) error {
 
 	return nil
 }
+
+func validateCephVersionsBetweenLocalAndExternalClusters(localVersion, externalVersion cephver.CephVersion, otherCRPresent, blockPoolsPresent bool) error {
+	// Identical version, regardless if other CRs are running, it's ok!
+	if cephver.IsIdentical(localVersion, externalVersion) {
+		return nil
+	}
+
+	// Versions are different and rgws / mds / nfs are running
+	// No block pools either
+	if !otherCRPresent && !blockPoolsPresent && !cephver.IsIdentical(localVersion, externalVersion) {
+		return fmt.Errorf("external cluster ceph version is different %s than the local cluster spec image %s and this is a new cluster", externalVersion.String(), localVersion.String())
+	}
+
+	// external version is 14.2.2 and local version is 14.2.1, and rgws / mds / nfs are running
+	// Can be higher from a major or minor standpoint
+	// Since other CRs are running it's acceptable but you should update soon!
+	if (otherCRPresent || blockPoolsPresent) && cephver.IsSuperior(externalVersion, localVersion) {
+		m := `external cluster ceph version is higher %s than the local cluster %s,
+			you have other CRDs running too and/or block pools created too, so consider updating your cluster cr image spec.`
+		logger.Warningf(m, externalVersion.String(), localVersion.String())
+		return nil
+
+	}
+
+	// external version is 14.2.2 and local version is 14.2.1, and rgws / mds / nfs are NOT running
+	// as well as no Block Pools have been created
+	// This assumes we just bootstrapped and it's not really acceptable
+	// So you should update the cluster CR image
+	if !otherCRPresent && !blockPoolsPresent && localVersion.Major == externalVersion.Major {
```

Comment:
Why do we need to restrict the version difference? I still believe it should only be a warning in the log. We're doing extra work to detect block pools and CRs for this corner case. Let's keep it simple and only error out if it's a different major version. If it's a different minor version write an error. Then we can simplify a lot in this PR since we wouldn't need to detect CRs or pools.